src/js/fable-core/Array2.fs

```diff
@@ -1,16 +1,732 @@
 module Array
 
+// This value is for use by compiled F# code and should not be used directly.
+// We can use LanguagePrimitives.ErrorStrings.
+// TODO: Approve please if we really need to use ErrorStrings here
+#nowarn "1204"
+
 open Fable.Core
+open Fable.Import
 
 type ArrayCons =
     [<Emit("new $0($1)")>]
     abstract Create: capacity: int -> 'T array
     [<Emit("$0.from($1)")>]
     abstract From: 'T seq -> 'T array
 
-let map (f: 'T->'U) (source: 'T[]) (cons: ArrayCons) =
+[<Emit("$0.push($1)")>]
+let private push (array: 'T[]) (item: 'T): int = jsNative
+
+[<Emit("$0.slice($1, $2)")>]
+let private slice (array: 'T[]) (``begin``: int) (``end``: int): 'T[] = jsNative
+
+[<Emit("$0.slice($1)")>]
+let private sliceFrom (array: 'T[]) (``begin``: int): 'T[] = jsNative
+
+[<Emit("$0.splice($1, $2)")>]
+let private splice (array: 'T[]) (start: int) (deleteCount: int): 'T[] = jsNative
+
+// TODO: Should we reuse declarations from Import.JS instead of raw emits?
+let inline private spliceN<'T> (array: 'T[]) (start: int) (deleteCount: int) =
+    (box array :?> JS.Array<_>).splice(float(start), float(deleteCount)) |> box :?> 'T[]
```

Comment:
Maybe using Import.JS can be a bit safer than just using JS code in a plain JS (and help us find problems with the bindings), but we should have the tests already so I don't really have a strong preference. If you prefer the emits because you can control the function signature that's fine with me :)

Reply:
I'll consider re-using `Fable.Import.JS` bindings then unless it hurts generated code just for the sake
 of avoiding duplicated definitions in different places 😃